modules/gdscript/gdscript_cache.cpp

```diff
@@ -151,22 +151,31 @@ void GDScriptCache::move_script(const String &p_from, const String &p_to) {
 		return;
 	}
 
-	if (singleton->parser_map.has(p_from) && !p_from.is_empty()) {
+	if (singleton->parser_map.has(p_from)) {
 		singleton->parser_map[p_to] = singleton->parser_map[p_from];
 	}
 	singleton->parser_map.erase(p_from);
 
-	if (singleton->parser_inverse_dependencies.has(p_from) && !p_from.is_empty()) {
+	if (singleton->dependencies.has(p_from)) {
+		singleton->dependencies[p_to] = singleton->dependencies[p_from];
+	}
+	singleton->dependencies.erase(p_from);
+
+	for (KeyValue<String, HashSet<String>> &E : singleton->dependencies) {
+		E.value.erase(p_from);
+	}
```

Comment:
Decided not to insert the new path, because then everything works fine even if the preloaded path in the script never changes, which is probably not desired